### PR TITLE
feat: FHB-967 Add KeyVault to Startup

### PR DESCRIPTION
**Pull request by StuwiiDev**
Added to SharedKernal and tested the Prefix.
Added to all projects that have appsettings that can pull from keyvault

### CODE REVIEW

File: src/shared/shared-kernel/src/fh-shared-kernel.shared-kernel/Extensions/StartupExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                new DefaultAzureCredential(),
+                new PrefixKeyVaultSecretManager(keyVaultEndpoint));
+        }
+    }
+
+    public sealed class PrefixKeyVaultSecretManager(string prefix) : KeyVaultSecretManager
+    {
+        private readonly string _prefix = $"{prefix}-";
+
+        // SD-API-CONNECTIONSTRINGS-SERVICEDIRECTORYCONNECTION
```
Comment (stevesatdfe) [inline src/shared/shared-kernel/src/fh-shared-kernel.shared-kernel/Extensions/StartupExtensions.cs:28]: Delete comment

**Review by stevesatdfe — DISMISSED** [review]
Just one commented out line which can be removed and fix the sudo issue - otherwise all looks fine

File: src/ui/connect-ui/tests/FamilyHubs.ReferralUi.UnitTests/Extensions/StartupExtensions/KeyVaultTests.cs
Comment (Zac-Digital) [inline src/ui/connect-ui/tests/FamilyHubs.ReferralUi.UnitTests/Extensions/StartupExtensions/KeyVaultTests.cs:1]: Should these tests not be in the Shared Kernel project given that's where the functionality is?

Comment (StuwiiDev) [reply src/ui/connect-ui/tests/FamilyHubs.ReferralUi.UnitTests/Extensions/StartupExtensions/KeyVaultTests.cs:1]: *face palm

**Review by stevesatdfe — APPROVED** [review]

**Review by Zac-Digital — APPROVED** [review]

**Pull request merged**